pipeline.cpp

```diff
@@ -80,7 +118,45 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
  ************************************************************/
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
-    // Your code goes here
+
+    //By finding the max and min of both are x and y axis we can find the outer box of the triangle. 
+    int maxX = fmax((int)triangle[0].x, fmax((int)triangle[1].x, (int)triangle[2].x));
```

Comment:
There is also the MAX3 function defined in definitions.  You got fmax to work fine here.